core/providers/node/package_manager.go

```diff
@@ -307,6 +324,11 @@ func (p PackageManager) GetPackageManagerPackages(ctx *generate.GenerateContext,
 	if p == PackageManagerBun {
 		bun := packages.Default("bun", "latest")
 
+		// Prefer explicit version from package.json engines over defaults
+		if packageJson != nil && packageJson.Engines != nil && packageJson.Engines["bun"] != "" {
+			packages.Version(bun, packageJson.Engines["bun"], "package.json > engines > bun")
+		}
```

Comment:
Nit: can this be extracted to something reusable since it is used in several place. Maybe something like